python/ray/util/spark/start_ray_node.py

```diff
@@ -109,11 +109,37 @@ def try_clean_temp_dir_at_exit():
             fcntl.flock(lock_fd, fcntl.LOCK_UN)
             os.close(lock_fd)
 
+    def check_parent_alive() -> None:
+        orig_parent_id = os.getppid()
```

Comment:
Actually there can be a race condition that when we execute this line, the parent process already died?

Should we just pass the ppid to the `start_ray_node` child process?

Reply:
This could be safer. Good suggestion.